<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="plot_cost" default="true"/>
    <arg name="plot_input" default="true"/>
    <arg name="plot_nominal_state" default="true"/>
    <arg name="enable_obstacle" default="false"/>

    <!-- Open Rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mppi_manipulation)/config/rviz/config.rviz" output="screen"/>

    <!--    Plot cost and other info-->
    <group if="$(arg plot_cost)">
        <arg name="mp_config" default="$(find mppi_panda)/resources/rqt_multiplot.xml"/>
        <node name="rqt_multiplot_cost" pkg="rqt_multiplot" type="rqt_multiplot"
              args="--multiplot-config $(arg mp_config)"
              output="screen"/>
    </group>

    <!--    Plot input-->
    <group if="$(arg plot_input)">
        <arg name="mp_config_input" default="$(find mppi_manipulation)/config/multiplot/rqt_multiplot.xml"/>
        <node name="rqt_multiplot_input" pkg="rqt_multiplot" type="rqt_multiplot"
              args="--multiplot-config $(arg mp_config_input)"
              output="screen"/>
    </group>

    <!--    Plot nominal base state -->
    <group if="$(arg plot_nominal_state)">
        <arg name="mp_config_nominal" default="$(find mppi_manipulation)/config/multiplot/base_nominal.xml"/>
        <node name="rqt_multiplot_nominal" pkg="rqt_multiplot" type="rqt_multiplot"
              args="--multiplot-config $(arg mp_config_nominal)"
              output="screen"/>
    </group>

    <!--        Interactive marker to command the end effector-->
    <node name="interactive_reference" pkg="mppi_panda" type="interactive_marker.py" output="screen">
        <param name="marker_server_name" value="reference_marker_server"/>
        <param name="target_frame" value="panda_grasp"/>
        <param name="frame_id" value="world"/>
        <param name="target_pose_topic" value="/end_effector_pose_desired"/>
    </node>

    <!--    Interactive marker for the end effector obstacle -->
    <group if="$(arg enable_obstacle)">
        <node name="interactive_obstacle" pkg="mppi_panda" type="interactive_marker.py" output="screen">
            <param name="marker_server_name" value="obstacle_marker_server"/>
            <param name="frame_id" value="world"/>
            <param name="target_pose_topic" value="/obstacle"/>
        </node>
    </group>
</launch>
